File: app/src/main/java/vn/loitp/app/activity/customviews/scratchview/scratchviewimage/ScratchViewImageActivity.java
package vn.loitp.app.activity.customviews.scratchview.scratchviewimage;

import android.app.Activity;
import android.os.Bundle;
import android.widget.TextView;

import loitp.basemaster.R;
import vn.loitp.core.base.BaseActivity;
import vn.loitp.views.scratchview.ScratchImageView;

public class ScratchViewImageActivity extends BaseActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ScratchImageView scratchImageView = (ScratchImageView) findViewById(R.id.sample_image);
        TextView textView = (TextView) findViewById(R.id.tv);
        scratchImageView.setRevealListener(new ScratchImageView.IRevealListener() {
            @Override
            public void onRevealed(ScratchImageView tv) {
                textView.setText("onRevealed");
            }

            @Override
            public void onRevealPercentChangedListener(ScratchImageView siv, float percent) {
                textView.setText("onRevealPercentChangedListener percent: " + percent);
            }
        });
    }

    @Override
    protected boolean setFullScreen() {
        return false;
    }

    @Override
    protected String setTag() {
        return getClass().getSimpleName();
    }

    @Override
    protected int setLayoutResourceId() {
        return R.layout.activity_scratchview_image;
    }
}

File: app/src/main/java/vn/loitp/app/activity/customviews/scratchview/scratchviewimage/ScratchViewImageActivity.kt
package vn.loitp.app.activity.customviews.scratchview.scratchviewimage

import android.annotation.SuppressLint
import android.os.Bundle
import com.annotation.IsFullScreen
import com.annotation.LogTag
import com.core.base.BaseFontActivity
import com.views.scratchview.LScratchImageView
import kotlinx.android.synthetic.main.activity_scratchview_image.*
import vn.loitp.app.R

@LogTag("ScratchViewImageActivity")
@IsFullScreen(false)
class ScratchViewImageActivity : BaseFontActivity() {

    override fun setLayoutResourceId(): Int {
        return R.layout.activity_scratchview_image
    }

    @SuppressLint("SetTextI18n")
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        scratchImageView.setRevealListener(object : LScratchImageView.IRevealListener {
            override fun onRevealed(iv: LScratchImageView) {
                textView.text = "onRevealed"
            }

            override fun onRevealPercentChangedListener(siv: LScratchImageView, percent: Float) {
                textView.text = "onRevealPercentChangedListener percent: $percent"
            }
        })
    }
}
